pkg/controller/identity/iampolicy/controller.go

```diff
@@ -99,7 +99,14 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	}
 
 	if meta.GetExternalName(cr) == "" {
-		return managed.ExternalObservation{}, nil
+		// If external name not set there is still a change it may already exist
+		// Try to get the policy by name
+		policyArn, _ := e.getPolicyArnByName(ctx, cr.Spec.ForProvider.Name)
+		if policyArn == nil {
+			return managed.ExternalObservation{}, nil
+		}
+		meta.SetExternalName(cr, aws.ToString(policyArn))
+		_ = e.kube.Update(ctx, cr)
```

Comment:
Often it causes issues if you do `kube.Update` inside a function like this, because the provider runtime also does it but with a different version number. Did you notice any errors in the log when you tested this?

Reply:
I did not see any errors in the logs